java/grpc-client/src/main/java/com/youtube/vitess/client/grpc/GrpcClientFactory.java

```diff
@@ -4,18 +4,257 @@
 import com.youtube.vitess.client.RpcClient;
 import com.youtube.vitess.client.RpcClientFactory;
 
+import io.grpc.netty.GrpcSslContexts;
 import io.grpc.netty.NegotiationType;
 import io.grpc.netty.NettyChannelBuilder;
+import io.netty.handler.ssl.SslContext;
 
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
 import java.net.InetSocketAddress;
+import java.security.*;
+import java.security.cert.*;
+import java.security.cert.Certificate;
+import java.util.Arrays;
+import java.util.Enumeration;
 
 /**
- * GrpcClientFactory creates RpcClients with the gRPC implemenation.
+ * GrpcClientFactory creates RpcClients with the gRPC implementation.
  */
 public class GrpcClientFactory implements RpcClientFactory {
+  /**
+   * TODO: Document
+   *
+   * @param ctx
+   * @param address
+   * @return
+   */
   @Override
   public RpcClient create(Context ctx, InetSocketAddress address) {
     return new GrpcClient(
-        NettyChannelBuilder.forAddress(address).negotiationType(NegotiationType.PLAINTEXT).build());
+            NettyChannelBuilder.forAddress(address).negotiationType(NegotiationType.PLAINTEXT).build());
   }
+
+  /**
+   * TODO: Document
+   *
+   * @param ctx
+   * @param address
+   * @return
+   */
+  public RpcClient createTls(Context ctx, InetSocketAddress address, TlsOptions tlsOptions) {
+    SslContext sslContext;
+    try {
+      final KeyStore keyStore = loadKeyStore(tlsOptions.getKeyStore(), tlsOptions.getKeyStorePassword());
+      final PrivateKeyWrapper privateKeyWrapper = tlsOptions.getKeyAlias() == null
+              ? loadPrivateKeyEntry(keyStore, tlsOptions.getKeyStorePassword(), tlsOptions.getKeyPassword())
+              : loadPrivateKeyEntryForAlias(keyStore, tlsOptions.getKeyAlias(), tlsOptions.getKeyStorePassword(), tlsOptions.getKeyPassword());
+
+      final KeyStore trustStore = loadKeyStore(tlsOptions.getTrustStore(), tlsOptions.getTrustStorePassword());
+      final X509Certificate[] trustCertCollection = {
+              (X509Certificate) trustStore.getCertificate(tlsOptions.getTrustAlias())
+      };
+
+      sslContext = GrpcSslContexts.forClient()
+              .keyManager(privateKeyWrapper.getPrivateKey(), privateKeyWrapper.getPassword(), privateKeyWrapper.getCertificateChain())
```

Comment:
Hi Harshit, my latest commit addresses all of the other comments you've raised... but I'm a bit unsure on how to proceed with this one.  

### Client vs. Server
I started down the path of initializing an `SslContext` object with `GrpcSslContexts.forClient()`, because it seemed like this code would always be a "client" context (i.e. its in `GrpcClientFactory` after all).  Are you saying that its possible to be in a "server" context here, and I should have some logical branching to detect that and build the `SslContext` differently for each branch?

### keyManager(...)
OR.... are you agreeing that this is indeed a "client" context, but are instead taking issue with the `keyManager(...)` call on the next line?  If so, then is the issue:

1. `keyManager` and `trustManager` can be optional (i.e. if VTGate is configured with certs that are publicly verifiable through Verisign or whoever, and need no special client config).  So don't call these two mutators if they keyStore and trustStore properties are `null`.
2. Or are you saying that if I call `keyManager`, then I should not also be using `trustManager`?  If so, then I'm totally lost.  :)

Or is the requested change something other than any of the possibilities above?  Thanks!